### ISSUE TITLE
Design for Accessibility Page

### ISSUE THREAD

**Issue by britanyponvelle** [IBM-Watson/design-library#140]
Visual design for Guidelines> Accessibility page.

**Comment by britanyponvelle** [comment]
Still working on the colors for the visual example. The orange is too much. 

![wdl-accessibility-large-06](https://cloud.githubusercontent.com/assets/9538528/6930723/5dc45130-d7cb-11e4-9a8e-300f230274c6.png)

**Comment by britanyponvelle** [comment]
![wdl-accessibility-large-v3](https://cloud.githubusercontent.com/assets/9538528/6931492/723c2ca2-d7d3-11e4-979a-b796582b16dd.png)

![resources-panel](https://cloud.githubusercontent.com/assets/9538528/6931493/73ed60de-d7d3-11e4-942e-b841998c14b2.png)

Trying out different things for the "resources" panel. Thoughts? @joshkimmell @poofichu

**Comment by poofichu** [comment]
@britanyponvelle I really like this.  It ties in very nicely with the current design language.  My personal preference is the one with the icon next to each link.

**Comment by britanyponvelle** [comment]
@joshkimmell: I added the word "links" because sometimes we will have links and downloads. If they are in the same box it could be confusing.

**Comment by britanyponvelle** [comment]
Update on resources panel:
![screen shot 2015-04-01 at 10 03 13 am](https://cloud.githubusercontent.com/assets/9538528/6943889/7cd8016c-d856-11e4-999b-c0bfaf6f505e.png)
![screen shot 2015-04-01 at 10 03 27 am](https://cloud.githubusercontent.com/assets/9538528/6943888/7cd7dc1e-d856-11e4-916c-251bbebda918.png)
![screen shot 2015-04-01 at 10 03 36 am](https://cloud.githubusercontent.com/assets/9538528/6943887/7cd7d3fe-d856-11e4-8882-412118e689a7.png)

**Comment by bobbinrobyn** [comment]
Option 2 is my vote - I think it does the best job of keeping the same type of items closer together. If there was an instance with no links, only downloads, would display in the left column?

**Comment by britanyponvelle** [comment]
That was my thought as well. Keeping the similar content together makes it easier to distinguish between the two. Although, I think keeping the panel full width looks more balanced:

![screen shot 2015-04-01 at 10 22 38 am](https://cloud.githubusercontent.com/assets/9538528/6944403/236aa884-d859-11e4-970a-411061db875b.png)
![screen shot 2015-04-01 at 10 23 14 am](https://cloud.githubusercontent.com/assets/9538528/6944402/2369f722-d859-11e4-94aa-805f54e4bd31.png)

Thoughts?

**Comment by joshkimmell** [comment]
I agree about the full width. Also, I like the categorical columns. I'm leaning toward a combination of options 2 & 3. Icons next to each resource without column headers, but keeping the resources in grouped columns.

**Comment by britanyponvelle** [comment]
Here is a combination of those two, using icons and color to differentiate links from files:

![screen shot 2015-04-01 at 10 47 30 am](https://cloud.githubusercontent.com/assets/9538528/6945051/be58dab6-d85c-11e4-8b58-ca14cd1969b8.png)

**Comment by britanyponvelle** [comment]
:+1: ?

**Comment by bobbinrobyn** [comment]
:+1: !

**Comment by ryanbrownhill** [comment]
:+1:

**Comment by britanyponvelle** [comment]
Updated: 
![wdl-accessibility-large-v4](https://cloud.githubusercontent.com/assets/9538528/6966738/dcc6aeaa-d920-11e4-9fe2-bebcdfe510e6.png)

**Comment by britanyponvelle** [comment]
http://pages.design.ibm.com/blponvel/ibm-watson-design-library/master/dist/accessibility.html

Accessibility coded!

**Comment by ryanbrownhill** [comment]
Updated content with animation accessibility

https://github.com/IBM-Watson/design-library/wiki/Accessibility-Guidelines

**Comment by kathrynmcelroy** [comment]
@ryanbrownhill Updated Accessibility text:

When choosing colors and contrast ratios for typography it is important to adhere to the IBM Design Language recommendations for accessibility, as we must meet WCAG 2.0 AA requirements. High color contrast helps users who are partially or completely color blind see differences between certain colors. It creates a strong visual hierarchy and improved usability for everyone. Never rely on color as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.


When designing animation it is also important to adhere to WCAG 2.0 AA requirements for accessibility. The visual design of the animated graphics should follow the guidelines set in the IBM Design Language Visual Section including: color contrast, descriptive labels, and meaningful text. In addition to the fundamental visual accessibility guides, animation has additional considerations. For animations that are longer than five seconds, we must provide the ability to pause, stop, or hide animations for users who can be severely distracted, particularly those with attention deficit disorders. The five-second limit is long enough to get a user’s attention, but short enough for a user to wait out the distraction before reading the page. (2.2b) Do not include anything that flashes or can be considered a flash more than two times in any one second for those prone to seizures. (2.3a) If the animation has audio longer than three seconds, provide audio control for users that have screen reading software, as it will make it easier for them to understand their screen reader. (1.4b)

**Comment by ryanbrownhill** [comment]
@kathrynmcelroy Updated the Wiki this your edits! Thanks!

**Comment by kathrynmcelroy** [comment]
My only other comment is that the Animation section repeats some of the Visual section, and adds more about visual that isn't in the first paragraph, specifically: "guidelines set in the IBM Design Language Visual Section including: color contrast, descriptive labels, and meaningful text."

I'm wondering if we should move that to the Visual paragraph, and then refer to it in the Animation paragraph.

**Comment by kathrynmcelroy** [comment]
One other edit on this sentence:

Do not include anything that flashes, or can be considered a flash, more than twice per second; it can negatively affect those prone to seizures.

**Comment by britanyponvelle** [comment]
@kathrynmcelroy I agree, we should move the content about general visual design to the top paragraph, and leave the content about animation in the bottom.

**Comment by britanyponvelle** [comment]
Here is updated content:

---

### Accessibility

#### Visual

Visual design should follow the guidelines set in the IBM Design Language Visual Section including: color contrast, descriptive labels, and meaningful text. When choosing colors and contrast ratios for typography it is important to adhere to the IBM Design Language recommendations for accessibility, as we must meet WCAG 2.0 AA requirements. High color contrast helps users who are partially or completely color-blind see differences between certain colors. It creates a strong visual hierarchy and improved usability for everyone. Never rely on color as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.  

![visual-accessibility](https://cloud.githubusercontent.com/assets/9538528/6948616/8f88b8b0-d874-11e4-8f40-f54cc582213f.png)  
**Distinguishing visual elements**
Help users to easily distinguish links and form elements by using combinations of colors, underlines, and iconography. 

**Resources:**  
1. [Color Contrast Analyzer](http://www.paciellogroup.com/resources/contrastanalyser/)  
2. [Contrast Ratio Analyzer](http://leaverou.github.io/contrast-ratio/)

#### Animation

In addition to the fundamental visual accessibility guides, animation has additional considerations. For animations that are longer than five seconds, we must provide the ability to pause, stop, or hide animations for users who can be severely distracted, particularly those with attention deficit disorders. The five-second limit is long enough to get a user’s attention, but short enough for a user to wait out the distraction before reading the page. [(2.2b)](http://w3-03.ibm.com/able/devtest/webpause.html) Do not include anything that flashes, or can be considered a flash, more than twice per second; it can negatively affect those prone to seizures. [(2.3a)](http://w3-03.ibm.com/able/devtest/webblinking.html) If the animation has audio longer than three seconds, provide audio control for users that have screen reading software, as it will make it easier for them to understand their screen reader. [(1.4b)](http://w3-03.ibm.com/able/devtest/webaudio.html)

Thoughts? If this looks good we can add it to the prototype.

**Comment by kathrynmcelroy** [comment]
Looks great! I have a couple tiny things:
- take out the colon after "including" (I originally put it there, but I don't think it needs one)
- take the dash out of "color blind"
- "In addition to the fundamental visual accessibility guides, animation has its own set of considerations."

**Comment by britanyponvelle** [comment]
Great, thanks @kathrynmcelroy ! I have updated all of the changes in the [wiki](https://github.com/IBM-Watson/design-library/wiki/Accessibility-Guidelines). Adding this to the prototype now!

**Comment by britanyponvelle** [comment]
I added the content to the prototype: http://pages.design.ibm.com/blponvel/ibm-watson-design-library/master/dist/accessibility.html

**Comment by britanyponvelle** [comment]
:+1: ?

**Comment by kathrynmcelroy** [comment]
There's one link that's messed up in the prototype (2.3a), but other than that: :+1:

**Comment by britanyponvelle** [comment]
We are deleting the links to the W3 page because 3rd party users cannot access them

**Comment by britanyponvelle** [comment]
@ryanbrownhill Josh wants to know if we can add a visual examples to the "Animation" example. Thoughts?

**Comment by ryanbrownhill** [comment]
I don't want to include "before / after" because it might cause seizures. The best example we can show is a pause, play controler for animations that last longer than 5 seconds.

**Comment by britanyponvelle** [comment]
updated visual accessibility example: http://pages.design.ibm.com/blponvel/ibm-watson-design-library/master/dist/accessibility.html

**Comment by britanyponvelle** [comment]
Updated the content for the accessibility example to mirror the Design Principles examples:
http://pages.design.ibm.com/blponvel/ibm-watson-design-library/master/dist/accessibility.html

**Comment by britanyponvelle** [comment]
Updated the content for the accessibility detail section to mirror the Design Principles detail section:
http://pages.design.ibm.com/blponvel/ibm-watson-design-library/master/dist/accessibility.html

**Comment by ryanbrownhill** [comment]
Animation example for accessibility - animations lasting longer than 5 seconds should have play pause control

![animation-accesibility](https://cloud.githubusercontent.com/assets/9538395/7185960/f2a9c548-e42b-11e4-82ff-c95c8659a6be.gif)

**Comment by britanyponvelle** [comment]
woohoo! Thanks @ryanbrownhill

**Comment by britanyponvelle** [comment]
This is done :+1:

**Comment by ryanbrownhill** [comment]
:+1: done 
Current Prototype
http://pages.design.ibm.com/blponvel/ibm-watson-design-library/master/dist/accessibility.html

**Comment by bobbinrobyn** [comment]
:+1:
